src/contexts/addresses.tsx

```diff
@@ -181,6 +271,7 @@ export const AddressesContextProvider: FC<{ overrideContextValue?: PartialDeep<A
 
```

Comment:
Mhh I was thinking about the fact that we store the `network` name in the address class... It's even used in the key identifying addresses in the local storage. What happens with the `custom` network, which could represent different networks? The network `name` could stay the same (=== custom) but the client (and the address amounts, tx etc.) could actually be different..
Could this be an issue?

Reply:
Yeah, this would be an issue if you are planning to use 2 different custom networks at the same time (same time = same login session). Need to think about it a bit and come back with a solution (even though this is an extreme corner case)

Reply:
My proposal:

The information of the addresses depends only on the blockchain (each address will have a different balance on `mainnet` than on `testnet` and on each of the infinite numbers of `custom` networks). So when we store information about an address in the state of the app, we need to associate this info with the blockchain that this info corresponds to. The solution I found so far was to simply store the _name_ of the blockchain/network(`['mainnet','testnet','local','custom']`) along with the address's information.

A better approach to be able to support the infinite number of custom blockchains would be to store the `nodeHost` URL instead of the network name. However, I think this is a bit of overengineering.

Our app's state only exists for as long as the user is logged in. It is flushed on user logout. Do we really see a use case where someone would need to use the wallet with 2 different custom settings within the same login session? In the extreme case that someone wants to do that (switch from one custom network to another custom network while being logged in) I propose that we only keep the state of the latest custom network, flushing and re-initializing every time the custom network changes. The only drawback of this approach would be that the pending transactions of the 1st custom network would not anymore be visible when switching back to it from the 2nd custom network. Everything else would work as expected.

Lemme know WYTAI (😆).

Reply:
I definitely see some cases where users would need to switch from one custom network to the other (developers during testing, miners etc.). I wouldn't say that it's an extreme case - even for everyday users - especially when you're trying to connect to other nodes if the one you usually use is down etc. I had to do exactly that one week ago with another wallet.

As for the drawback you described - it's good enough for now.
Let's keep it like that and see if some people are complaining down the line.

Thanks for taking the time to investigate :)